src/coreclr/tools/Common/TypeSystem/Common/TargetDetails.cs

```diff
@@ -109,21 +109,21 @@ public int MaximumAlignment
             {
                 if (Architecture == TargetArchitecture.ARM)
                 {
-                    // Corresponds to alignment required for __m128 (there's no __m256)
+                    // Corresponds to alignment required for __m128 (there's no __m256/__m512)
                     return 8;
                 }
                 else if (Architecture == TargetArchitecture.ARM64)
                 {
-                    // Corresponds to alignmet required for __m256
+                    // Corresponds to alignmet required for __m128 (there's no __m256/__m512)
                     return 16;
                 }
                 else if (Architecture == TargetArchitecture.LoongArch64)
                 {
                     return 16;
                 }
 
-                // 256-bit vector is the type with the highest alignment we support
-                return 32;
+                // 512-bit vector is the type with the highest alignment we support
```

Comment:
This is an R2R breaking change. You need to update the R2R major/minor/minimum version.

Reply:
@davidwrighton (or @MichalStrehovsky?) does the major have to be updated here?

This is a "net new" type, so it seems somewhat unexpected that we'd need to bump from 8.0 to 9.0 just because of that.

Reply:
CC. @trylek, you might also be able to assist here

Reply:
Well, if I'm not mistaken, the alignment / packing change has the potential to change type layouts; that means that this change requires a counterpart CoreCLR runtime change (that seems to be included in your PR), however the two runtimes cannot accept the same R2R version because each of them calculates a different type layout at runtime so one of them will be mismatched with the R2R version.

Similarly, an app compiled with Crossgen2 before and after this change cannot be run by the same version of the CoreCLR runtime because the two versions of Crossgen2 potentially calculate different type layouts and one of them will be mismatched with the CoreCLR runtime. For this reason I believe that David is right that the major R2R format version must be incremented because that's exactly what's used for matching the runtime version with version of the R2R-compiled code. 

We wouldn't have this problem if we were storing type layouts in the R2R file, then the runtime wouldn't need to calculate them by itself, but the problem wouldn't go away altogether (e.g. within a larger project you still must compile all assemblies using the same version of Crossgen (not half of the files with Crossgen before this change and ther other half using Crossgen after this change) because they could have different internal views of the same type.

I haven't yet managed to examine your PR in its entirety so maybe there's some aspect I'm missing, let me know if my arguments make sense to you.

Reply:
Makes sense. Will update

Reply:
Possibly you could make this change partially version-neutral by e.g. only bumping up the alignment and packing size for types containing an instance field of the new Vector512 type; that would mean that code compiled with older Crossgen2 would still work with the newer CoreCLR runtime and code compiled with the new Crossgen2 but not using the new type would work on the older runtime but I guess at some point you still need to differentiate between the runtime that has this change and the runtime that doesn't as they're simply capable of running different code.

Reply:
Resolved